### PR TITLE
Update data format, apply some style changes, enable adding sessions

### LINKED ISSUE #92: Fix session creation
There are problems with `db.add_session`:
- `Session.to_dict` fails when there is no volume (recall from 87c4482d37a7e722656aa3c21a04784a448204d4 that we wanted to phase out `volume` in favor of `volume_id` – so apparently that transition needs to be taken to completion). 
- Serializing a session results in duplication of the entire transducer geometry data. So if you start with a session that defines a transducer by referencing its ID, then when you serialize that you end up with a session that defines a transducer by defining its geometry. The serialization/deserialization of Sessions needs to be cleaned up like we did for Protocols and Transducers in #85 
- There is no check that the subject ID inside the session json is the same as the the subject specified in the call to `db.add_session`

### LINKED ISSUE #97: Remove matrix from transducer definition
Transducers should not have a transform matrix -- this should instead be saved into sessions.
There are functions like `run_simulation` that use the matrix defined inside a `Transducer` so this would have to be re-designed a bit.

### LINKED ISSUE #107: getsiscale does not parse frequencies correctly
`openlifu.utils.units.getsiscale` does not parse these frequencies correctly:

MHz
GHz
THz

### PR DESCRIPTION
**Pull request by ebrahimebrahim**
Updates the session and transducer data format in a few ways.

Also takes this opportunity to relax linting exceptions and apply some of the more sweeping style changes. This will help streamline future development.

Close #92 
Make some progress towards #106 (but we can leave it open to continue chipping away at that)
Close #107
Close #97

### CODE REVIEW

**Comment by ebrahimebrahim** [comment]
Hi @medmssaouri, this PR isn't as large as it looks, the sweeping changes across many files are due to the "relax linting exceptions" commit that makes only style changes. Please have a look one commit at a time and you can ignore the style commit, then it's not so many changes. In case you want to test, this branch should work with the "v02" data archives here: https://drive.google.com/drive/folders/1bldoP9VxE9SHyAB0bVAJ2l6WUuTxz3Yu?usp=drive_link

**Comment by ebrahimebrahim** [comment]
actually the only example data it currently works with is `db-simple-example-v03.zip`

the db-extended-example we will not become compatible with until after #93

File: tests/test_database.py [hunk 1]
```diff
@@ -12,15 +13,28 @@ def example_database(tmp_path:Path) -> Database:
     shutil.copytree(Path(__file__).parent/'resources/example_db', tmp_path/"example_db")
     return Database(tmp_path/"example_db")
 
-
-
-def test_load_session_from_file(example_database : Database):
-    session = Session.from_file(
+@pytest.fixture()
```
Comment (medmssaouri) [inline tests/test_database.py:16]: If i understood well the goal of the commit [Remove transducer in favor of only ID in Session](https://github.com/OpenwaterHealth/OpenLIFU-python/pull/110/commits/c50d98e85b7359fa71688cd7a67fa1b5d761ef08) i think that it would be better to move helper and session tests to separate commits and to squash helper commit with [Simplify serialization test code](https://github.com/OpenwaterHealth/OpenLIFU-python/pull/110/commits/b23ff826524e8518ced86591fe48c3e4ce6bc210)

File: src/openlifu/db/session.py [hunk 2]
```diff
@@ -152,5 +153,6 @@ def to_file(self, filename):
 
         :param filename: Name of the file
         """
+        Path(filename).parent.mkdir(exist_ok=True)
```
Comment (medmssaouri) [inline src/openlifu/db/session.py:156]: I think that this should be in a separate commit (or maybe modifying commit message to include fix to_file)

**Comment by medmssaouri** [comment]
@ebrahimebrahim  Thanks, I’ve reviewed it and made some minor suggestions.

File: src/openlifu/bf/__init__.py [hunk 3]
```diff
@@ -1,5 +1,15 @@
-from .delay_methods import DelayMethod
 from .apod_methods import ApodizationMethod
+from .delay_methods import DelayMethod
 from .focal_patterns import FocalPattern, SinglePoint, Wheel
 from .pulse import Pulse
 from .sequence import Sequence
+
+__all__ = [
```
[the remark this answers is not in the document]
Comment (ebrahimebrahim) [reply src/openlifu/bf/__init__.py:7]: Thanks, I will edit the commit message. It's part of "remove unused imports" because adding things to `__all__` is a way to indicate the intention that something is being imported in order to be exported, which is needed for the linting to pass after removal of the unused-import exception

File: tests/test_units.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import pytest
+import numpy as np
+from openlifu.util.units import (
+    getsiscale
+)
+
+def test_getsiscale():
```
[the remark this answers is not in the document]
Comment (ebrahimebrahim) [reply tests/test_units.py:7]: In general if it's not obvious what's being tested it's good to include a docstring. I just imagine "if this test fails one year later after i've completely forgotten about the project, will I understand what's going on?" :smile:

File: tests/test_database.py (hunk 1 above)
Comment (ebrahimebrahim) [reply tests/test_database.py:16]: Hmm but the unit tests being added here are testing the code changes that are made in the very same commit. I feel it should stay like this

File: src/openlifu/db/session.py (hunk 2 above)
Comment (ebrahimebrahim) [reply src/openlifu/db/session.py:156]: Good point, thanks

File: tests/test_units.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+    assert getsiscale('mm', 'distance') == 1e-3
+    assert getsiscale('km', 'distance') == 1e3
+    assert getsiscale('mm^2', 'area') == 1e-6
+    assert getsiscale('mm^3', 'volume') == 1e-9
+    assert getsiscale('ns', 'time') == 1e-9
+    assert getsiscale('nanosecond', 'time') == 1e-9
+    assert getsiscale('hour', 'time') == 3600.
+    assert getsiscale('rad', 'angle') == 1.
+    assert np.allclose(getsiscale('deg', 'angle'), np.pi/180)
```
[the remark this answers is not in the document]
Comment (medmssaouri) [reply tests/test_units.py:19]: It's seems that this one still pending

**Comment by medmssaouri** [comment]
LGTM once the last unresolved conversation is fixed, i will test once after #93

File: tests/test_units.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (ebrahimebrahim) [reply tests/test_units.py]: Oh oops I thought I had resolved it. Now it shoudl be good

**Review by medmssaouri — APPROVED** [review]
LGTM !!
